src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/IconUrlToImageCacheConverter.cs

```diff
@@ -86,49 +77,52 @@ public object Convert(object[] values, Type targetType, object parameter, Cultur
+                        if (reader is PackageArchiveReader par) // This reader is closed in BitmapImage events
                         {
-                            par.Dispose();
-                            IconBitmapImage_DownloadOrDecodeFailed(sender, args);
-                            AddToCache(iconUrl, defaultPackageIcon);
-                        };
-                        iconBitmapImage.DownloadFailed += (sender, args) =>
+                            iconBitmapImage.StreamSource = par.GetEntry(iconEntry).Open();
+
+                            iconBitmapImage.DecodeFailed += (sender, args) =>
+                            {
+                                reader.Dispose();
+                                IconBitmapImage_DownloadOrDecodeFailed(sender, args);
+                                AddToCache(iconUrl, defaultPackageIcon);
+                            };
+                            iconBitmapImage.DownloadFailed += (sender, args) =>
+                            {
+                                reader.Dispose();
+                                IconBitmapImage_DownloadOrDecodeFailed(sender, args);
+                                AddToCache(iconUrl, defaultPackageIcon);
+                            };
+                            iconBitmapImage.DownloadCompleted += (sender, args) =>
+                            {
+                                reader.Dispose();
+                                IconBitmapImage_DownloadCompleted(sender, args);
+                            };
+
+                            imageResult = FinishImageProcessing(iconBitmapImage, iconUrl, defaultPackageIcon);
+                        }
+                        else // we cannot use the reader object
                         {
-                            par.Dispose();
-                            IconBitmapImage_DownloadOrDecodeFailed(sender, args);
                             AddToCache(iconUrl, defaultPackageIcon);
-                        };
-                        iconBitmapImage.DownloadCompleted += (sender, args) =>
-                        {
-                            par.Dispose();
-                            IconBitmapImage_DownloadCompleted(sender, args);
-                        };
-
-                        imageResult = FinishImageProcessing(iconBitmapImage, iconUrl, defaultPackageIcon);
+                            imageResult = defaultPackageIcon;
+                        }
                     }
                     catch (Exception)
                     {
                         AddToCache(iconUrl, defaultPackageIcon);
                         imageResult = defaultPackageIcon;
                     }
                 }
-                // Identified an embedded icon URI but, we are unable to process it
-                // cache and return the default image
-                else
+                else // Identified an embedded icon URI but, we are unable to process it
```

Comment:
`...URI, but we...`

The comma is in the wrong place.

Reply:
Fixed.